cpp/oneapi/dal/backend/primitives/reduction/test/reduction_rm_random_dpc.cpp

```diff
@@ -251,6 +259,54 @@ class reduction_rm_test_random : public te::float_algo_fixture<std::tuple_elemen
     table input_table_;
 };
 
+template <typename Param>
+class infinite_sum_rm_test_random : public reduction_rm_test_random<Param> {
+public:
+    using float_t = std::tuple_element_t<0, Param>;
+    using binary_t = std::tuple_element_t<1, Param>;
+    using unary_t = std::tuple_element_t<2, Param>;
+
+    void generate(bool maxval) {
+        this->width_ = GENERATE(7, 707, 5);
+        this->stride_ = GENERATE(707, 812, 1024);
+        this->height_ = GENERATE(17, 999, 1, 1001);
+        CAPTURE(this->width_, this->stride_, this->height_);
+        generate_input(maxval);
+    }
+
+    void generate_input(bool maxval) {
+        float_t inp = 0.9 * (float_t)maxval * std::numeric_limits<float_t>::max() + 0.5;
+        const auto train_dataframe = GENERATE_DATAFRAME(
+            te::dataframe_builder{ this->height_, this->stride_ }.fill_uniform(-0.2, inp));
+        this->input_table_ = train_dataframe.get_table(this->get_homogen_table_id());
+    }
+};
+
+template <typename Param>
+class single_infinite_rm_test_random : public reduction_rm_test_random<Param> {
+public:
+    using float_t = std::tuple_element_t<0, Param>;
+    using binary_t = std::tuple_element_t<1, Param>;
+    using unary_t = std::tuple_element_t<2, Param>;
+    void generate(bool infval) {
+        this->width_ = GENERATE(7, 707, 5);
+        this->stride_ = GENERATE(707, 812, 1024);
+        this->height_ = GENERATE(17, 999, 1, 1001);
+        CAPTURE(this->width_, this->stride_, this->height_);
+        generate_input(infval);
+    }
+
+    void generate_input(bool infval) {
+        const auto train_dataframe = GENERATE_DATAFRAME(
+            te::dataframe_builder{ this->height_, this->stride_ }.fill_uniform(-0.2, 0.5));
```

Comment:
why do you use stride instead of width?

Reply:
Its what is used above in the same file, I thought it was weird too, but didn't want to go too far away from what has already been implemented.

Reply:
Yes, I think its a bug, can you please try to check that everything is fine, if you provide width